connector.py", line 309, in dataloader
    return call._call_lightning_datamodule_hook(self.instance.trainer, self.name)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/call.py", line 1
79, in _call_lightning_datamodule_hook
    return fn(*args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/core/hooks.py", line 510
, in test_dataloader
    raise MisconfigurationException("`test_dataloader` must be implemented to be used with the Lightning Trainer")
lightning_fabric.utilities.exceptions.MisconfigurationException: `test_dataloader` must be implemented to be used with t
he Lightning Trainer

2024-09-13 06:50:57.486 | INFO     | __main__:main:1368 - length of lean_git_repos: 1
2024-09-13 06:50:57.486 | INFO     | __main__:main:1369 - i: 1
2024-09-13 06:50:57.486 | INFO     | __main__:main:1692 - Finished processing the repository
2024-09-13 06:50:57.487 | INFO     | __main__:main:1696 - An error occurred: list index out of range
2024-09-13 06:50:57.487 | INFO     | __main__:main:1694 - current epoch: 2
Traceback (most recent call last):
  File "/data/yingzi_ma/lean_project/ReProver/main.py", line 1370, in main
    repo = lean_git_repos[i]
IndexError: list index out of range
2024-09-13 06:50:57.487 | INFO     | __main__:main:1368 - length of lean_git_repos: 1
2024-09-13 06:50:57.487 | INFO     | __main__:main:1369 - i: 1
2024-09-13 06:50:57.487 | INFO     | __main__:main:1696 - An error occurred: list index out of range
Traceback (most recent call last):
  File "/data/yingzi_ma/lean_project/ReProver/main.py", line 1370, in main
    repo = lean_git_repos[i]
IndexError: list index out of range
[rank0]:[W913 06:51:01.716522428 ProcessGroupNCCL.cpp:1168] Warning: WARNING: process group has NOT been destroyed befor
e we destruct ProcessGroupNCCL. On normal program exit, the application should call destroy_process_group to ensure that
 any pending NCCL operations have finished in this process. In rare cases this process can exit before this point and bl
ock the progress of another member of the process group. This constraint has always been present,  but this warning has
only been added since PyTorch 2.4 (function operator())
^Z
[6]+  Stopped                 bash run_code.sh
(base) yingzi_ma@compute-permanent-node-352:~/lean_project/ReProver$ bash run_code.sh
Script executed from: /data/yingzi_ma/lean_project/ReProver
Removing old cache files
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834741.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834743.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834746.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834747.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_835354.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_GCS.log': Permission denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834744.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_RAYLET.log': Permission denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834742.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_833907.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_AUTOSCALER.log': Permission deni
ed
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834748.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834745.log': Permiss
ion denied
Stopping ray
^Z
[7]+  Stopped                 bash run_code.sh
(base) yingzi_ma@compute-permanent-node-352:~/lean_project/ReProver$ nvidia-smi
Fri Sep 13 06:57:20 2024
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 550.90.07              Driver Version: 550.90.07      CUDA Version: 12.4     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA A100-SXM4-80GB          On  |   00000000:50:00.0 Off |                    0 |
| N/A   38C    P0             88W /  400W |   23389MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   1  NVIDIA A100-SXM4-80GB          On  |   00000000:8C:00.0 Off |                    0 |
| N/A   40C    P0             87W /  400W |   16139MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   2  NVIDIA A100-SXM4-80GB          On  |   00000000:91:00.0 Off |                    0 |
| N/A   37C    P0             89W /  400W |   16139MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   3  NVIDIA A100-SXM4-80GB          On  |   00000000:D6:00.0 Off |                    0 |
| N/A   38C    P0             84W /  400W |   15995MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A   2277299      C   python                                      16840MiB |
|    0   N/A  N/A   2277848      C   ...miniconda3/envs/ReProver/bin/python       1348MiB |
|    0   N/A  N/A   2277849      C   ...miniconda3/envs/ReProver/bin/python       1348MiB |
|    0   N/A  N/A   2277850      C   ...miniconda3/envs/ReProver/bin/python       1348MiB |
|    0   N/A  N/A   2280297      C   python                                       2482MiB |
|    1   N/A  N/A   2277848      C   ...miniconda3/envs/ReProver/bin/python      16132MiB |
|    2   N/A  N/A   2277849      C   ...miniconda3/envs/ReProver/bin/python      16132MiB |
|    3   N/A  N/A   2277850      C   ...miniconda3/envs/ReProver/bin/python      15988MiB |
+-----------------------------------------------------------------------------------------+
(base) yingzi_ma@compute-permanent-node-352:~/lean_project/ReProver$ python ppl_gpus.py
GPU UUID                                      PID  User       Process Name      GPU Memory Usage (MB)    GPU Index  GPU
Name
----------------------------------------  -------  ---------  --------------  -----------------------  -----------  ----
-----------------
GPU-51be0465-4582-0cee-c9ea-313e1da175e4  2277299  yingzi_ma  pt_main_thread                    16840            0  NVID
IA A100-SXM4-80GB
GPU-51be0465-4582-0cee-c9ea-313e1da175e4  2277848  yingzi_ma  pt_main_thread                     1348            0  NVID
IA A100-SXM4-80GB
GPU-51be0465-4582-0cee-c9ea-313e1da175e4  2277850  yingzi_ma  pt_main_thread                     1348            0  NVID
IA A100-SXM4-80GB
GPU-51be0465-4582-0cee-c9ea-313e1da175e4  2277849  yingzi_ma  pt_main_thread                     1348            0  NVID
IA A100-SXM4-80GB
GPU-51be0465-4582-0cee-c9ea-313e1da175e4  2280297  yingzi_ma  pt_main_thread                     2482            0  NVID
IA A100-SXM4-80GB
GPU-70e95e0c-af26-5246-ffbe-af3ba17311f0  2277848  yingzi_ma  pt_main_thread                    16132            1  NVID
IA A100-SXM4-80GB
GPU-e9616018-dcca-df1c-3546-b4d5fb43d9c1  2277849  yingzi_ma  pt_main_thread                    16132            2  NVID
IA A100-SXM4-80GB
GPU-d9d07267-4ff6-30d9-36d3-c508cbf837b8  2277850  yingzi_ma  pt_main_thread                    15988            3  NVID
IA A100-SXM4-80GB
Killed process 2277299
Killed process 2277848
Killed process 2277850
Killed process 2277849
Killed process 2280297
Killed process 2277848
Killed process 2277849
Killed process 2277850
(base) yingzi_ma@compute-permanent-node-352:~/lean_project/ReProver$ nvidia-smi
Fri Sep 13 06:57:24 2024
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 550.90.07              Driver Version: 550.90.07      CUDA Version: 12.4     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA A100-SXM4-80GB          On  |   00000000:50:00.0 Off |                    0 |
| N/A   38C    P0             88W /  400W |   20901MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   1  NVIDIA A100-SXM4-80GB          On  |   00000000:8C:00.0 Off |                    0 |
| N/A   40C    P0             87W /  400W |   16139MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   2  NVIDIA A100-SXM4-80GB          On  |   00000000:91:00.0 Off |                    0 |
| N/A   37C    P0             89W /  400W |   16139MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   3  NVIDIA A100-SXM4-80GB          On  |   00000000:D6:00.0 Off |                    0 |
| N/A   38C    P0             83W /  400W |   15995MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
+-----------------------------------------------------------------------------------------+
(base) yingzi_ma@compute-permanent-node-352:~/lean_project/ReProver$ python ppl_gpus.py
GPU UUID                                      PID  User       Process Name      GPU Memory Usage (MB)    GPU Index  GPU
Name
----------------------------------------  -------  ---------  --------------  -----------------------  -----------  ----
-----------------
GPU-51be0465-4582-0cee-c9ea-313e1da175e4  2277299  yingzi_ma  pt_main_thread                    16840            0  NVID
IA A100-SXM4-80GB
Killed process 2277299
(base) yingzi_ma@compute-permanent-node-352:~/lean_project/ReProver$ nvidia-smi
Fri Sep 13 06:57:28 2024
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 550.90.07              Driver Version: 550.90.07      CUDA Version: 12.4     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA A100-SXM4-80GB          On  |   00000000:50:00.0 Off |                    0 |
| N/A   38C    P0             88W /  400W |   20901MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   1  NVIDIA A100-SXM4-80GB          On  |   00000000:8C:00.0 Off |                    0 |
| N/A   40C    P0             87W /  400W |   16139MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   2  NVIDIA A100-SXM4-80GB          On  |   00000000:91:00.0 Off |                    0 |
| N/A   37C    P0             89W /  400W |   16139MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   3  NVIDIA A100-SXM4-80GB          On  |   00000000:D6:00.0 Off |                    0 |
| N/A   38C    P0             84W /  400W |   15995MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
+-----------------------------------------------------------------------------------------+
(base) yingzi_ma@compute-permanent-node-352:~/lean_project/ReProver$ nvidia-smi
Fri Sep 13 06:57:30 2024
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 550.90.07              Driver Version: 550.90.07      CUDA Version: 12.4     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA A100-SXM4-80GB          On  |   00000000:50:00.0 Off |                    0 |
| N/A   38C    P0             88W /  400W |   20901MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   1  NVIDIA A100-SXM4-80GB          On  |   00000000:8C:00.0 Off |                    0 |
| N/A   40C    P0             87W /  400W |   16139MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   2  NVIDIA A100-SXM4-80GB          On  |   00000000:91:00.0 Off |                    0 |
| N/A   37C    P0             89W /  400W |   16139MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   3  NVIDIA A100-SXM4-80GB          On  |   00000000:D6:00.0 Off |                    0 |
| N/A   38C    P0             83W /  400W |   15995MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
+-----------------------------------------------------------------------------------------+
(base) yingzi_ma@compute-permanent-node-352:~/lean_project/ReProver$ bash run_code.sh
Script executed from: /data/yingzi_ma/lean_project/ReProver
Removing old cache files
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834741.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834743.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834746.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834747.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_835354.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_GCS.log': Permission denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834744.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_RAYLET.log': Permission denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834742.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_833907.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_AUTOSCALER.log': Permission deni
ed
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834748.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834745.log': Permiss
ion denied
Stopping ray
Could not terminate `/usr/bin/python3 /usr/local/bin/user_traffic_monitor.py` due to (pid=784996, name='python3')
Did not find any active Ray processes.
Running main.py
[2024-09-13 06:57:55,878] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cuda (auto detect)
2024-09-13 06:58:04.863 | INFO     | __main__:main:1197 - Running progressive training
2024-09-13 06:58:04.863 | INFO     | __main__:main:1203 - Configuring LeanDojo...
2024-09-13 06:58:04.866 | INFO     | generate_benchmark_lean4:configure_leandojo:347 - Current working directory: /data/
yingzi_ma/lean_project/ReProver
2024-09-13 06:58:04.866 | INFO     | __main__:main:1205 - LeanDojo configured
2024-09-13 06:58:04.866 | INFO     | __main__:main:1210 - Starting the main process
2024-09-13 06:58:04.866 | INFO     | __main__:main:1218 - Loading database from /data/yingzi_ma/lean_project/dynamic_dat
abase_PT_single_repo_no_ewc.json
2024-09-13 06:58:25.811 | INFO     | dynamic_database:add_repository:600 - Attempting to add repository: https://github.
com/leanprover-community/mathlib4 (commit: 2b29e73438e240a427bcecc7c0fe19306beb1310)
2024-09-13 06:58:25.811 | INFO     | dynamic_database:add_repository:603 - Added new repository: https://github.com/lean
prover-community/mathlib4 (commit: 2b29e73438e240a427bcecc7c0fe19306beb1310)
2024-09-13 06:58:29.762 | INFO     | dynamic_database:add_repository:600 - Attempting to add repository: https://github.
com/lecopivo/SciLean (commit: 22d53b2f4e3db2a172e71da6eb9c916e62655744)
2024-09-13 06:58:29.763 | INFO     | dynamic_database:add_repository:603 - Added new repository: https://github.com/leco
pivo/SciLean (commit: 22d53b2f4e3db2a172e71da6eb9c916e62655744)
2024-09-13 06:58:30.054 | INFO     | __main__:main:1220 - Loaded database from /data/yingzi_ma/lean_project/dynamic_data
base_PT_single_repo_no_ewc.json
2024-09-13 06:58:30.054 | INFO     | __main__:main:1227 - Found 15 repositories
2024-09-13 06:58:30.055 | INFO     | __main__:main:1318 - Starting without curriculum learning
2024-09-13 06:58:30.423 | DEBUG    | lean_dojo.data_extraction.lean:_to_commit_hash:90 - Querying the commit hash for le
an4 v4.7.0
2024-09-13 06:58:46.324 | INFO     | __main__:main:1347 - Finding compatible repositories...
2024-09-13 06:58:46.339 | INFO     | __main__:main:1350 - Finished finding compatible repositories
2024-09-13 06:58:46.339 | INFO     | __main__:main:1368 - length of lean_git_repos: 1
2024-09-13 06:58:46.339 | INFO     | __main__:main:1369 - i: 0
2024-09-13 06:58:46.340 | INFO     | __main__:main:1375 - Main process
2024-09-13 06:58:46.340 | INFO     | __main__:main:1376 - Using lambda = 0.1
2024-09-13 06:58:46.340 | INFO     | __main__:main:1377 - Processing https://github.com/lecopivo/SciLean
2024-09-13 06:58:46.340 | INFO     | __main__:main:1398 - Adding repo to repos_for_merged_dataset
2024-09-13 06:58:46.340 | INFO     | __main__:main:1409 - All GPUs
2024-09-13 06:58:46.342 | INFO     | __main__:find_latest_checkpoint:517 - Using the latest checkpoint: /data/yingzi_ma/
lean_project/checkpoints_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0
.1_epoch=1-Recall@10_val=70.07.ckpt
2024-09-13 06:58:46.342 | INFO     | __main__:main:1416 - Found latest checkpoint: /data/yingzi_ma/lean_project/checkpoi
nts_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@10_
val=70.07.ckpt
2024-09-13 06:58:46.342 | INFO     | __main__:main:1422 - Inside train_test_fisher
2024-09-13 06:58:46.342 | INFO     | __main__:main:1423 - Starting training at epoch 1
[rank: 0] Seed set to 3407
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/lightning_fabric/utilities/cloud_io.py:57: You are
 using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicit
ly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://g
ithub.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default val
ue for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arb
itrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user v
ia `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you
 don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental
feature.
2024-09-13 06:58:48.949 | INFO     | __main__:main:1447 - Loaded premise retriever at /data/yingzi_ma/lean_project/check
points_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@
10_val=70.07.ckpt
2024-09-13 06:58:48.952 | INFO     | __main__:find_latest_fisher:527 - Using the latest Fisher Information Matrix: /data
/yingzi_ma/lean_project/fisher_PT_single_repo_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_fishe
r_info.pkl
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/storage.py:414: FutureWarning: You are using
 `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It
 is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.
com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for
 `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary
 objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `to
rch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't
 have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental featur
e.
  return torch.load(io.BytesIO(b))
2024-09-13 06:58:49.288 | INFO     | __main__:load_fisher_information:503 - Fisher Information successfully loaded.
2024-09-13 06:58:49.288 | INFO     | retrieval.model:set_fisher_info:60 - Fisher Information has been updated in the mod
el.
2024-09-13 06:58:49.288 | INFO     | __main__:main:1454 - Fisher Information Matrix loaded.
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/lightning_fabric/plugins/environments/slurm.py:204
: The `srun` command is available on your system but is not used. HINT: If your intention is to run Lightning on SLURM,
prepend your python command with `srun` like so: srun python main.py ...
Using bfloat16 Automatic Mixed Precision (AMP)
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
`Trainer(limit_train_batches=1.0)` was configured so 100% of the batches per epoch will be used..
`Trainer(limit_val_batches=1.0)` was configured so 100% of the batches will be used..
2024-09-13 06:58:49.646 | INFO     | __main__:main:1507 - right before barrier for data module
2024-09-13 06:58:49.647 | INFO     | __main__:main:1522 - Data path: /data/yingzi_ma/lean_project/datasets_PT_single_rep
o_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: Futu
reWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be deprac
ted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github
.com/huggingface/transformers/issues/31884
  warnings.warn(
2024-09-13 06:58:49.765 | INFO     | common:__init__:212 - Building the corpus from /data/yingzi_ma/lean_project/dataset
s_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/corpus.jsonl
2024-09-13 06:58:58.927 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
Training dataset size: 149228
2024-09-13 06:58:58.943 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_val/cached_data.pkl
Validation dataset size: 2087
2024-09-13 06:58:58.957 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_pred/cached_data.pkl
Testing dataset size: 2298
2024-09-13 06:58:58.957 | INFO     | __main__:main:1536 - Training dataset size after load: 149228
2024-09-13 06:58:58.957 | INFO     | __main__:main:1537 - Validation dataset size after load: 2087
2024-09-13 06:58:58.957 | INFO     | __main__:main:1538 - Testing dataset size after load: 2298
2024-09-13 06:58:58.957 | INFO     | __main__:main:1540 - Starting progressive training from epoch 1 to 2
2024-09-13 06:58:58.957 | INFO     | __main__:main:1543 - hit the barrier before training
[rank: 0] Seed set to 3407
Initializing distributed: GLOBAL_RANK: 0, MEMBER: 1/4
[2024-09-13 06:59:10,013] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2024-09-13 06:59:10,017] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2024-09-13 06:59:10,017] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cuda (auto detect)
2024-09-13 06:59:18.455 | INFO     | __main__:main:1197 - Running progressive training
2024-09-13 06:59:18.455 | INFO     | __main__:main:1197 - Running progressive training
2024-09-13 06:59:18.455 | INFO     | __main__:main:1197 - Running progressive training
2024-09-13 06:59:18.455 | INFO     | __main__:main:1203 - Configuring LeanDojo...
2024-09-13 06:59:18.455 | INFO     | __main__:main:1203 - Configuring LeanDojo...
2024-09-13 06:59:18.455 | INFO     | __main__:main:1203 - Configuring LeanDojo...
2024-09-13 06:59:18.459 | INFO     | generate_benchmark_lean4:configure_leandojo:347 - Current working directory: /data/
yingzi_ma/lean_project/ReProver
2024-09-13 06:59:18.459 | INFO     | generate_benchmark_lean4:configure_leandojo:347 - Current working directory: /data/
yingzi_ma/lean_project/ReProver
2024-09-13 06:59:18.459 | INFO     | generate_benchmark_lean4:configure_leandojo:347 - Current working directory: /data/
yingzi_ma/lean_project/ReProver
2024-09-13 06:59:18.459 | INFO     | __main__:main:1205 - LeanDojo configured
2024-09-13 06:59:18.459 | INFO     | __main__:main:1205 - LeanDojo configured
2024-09-13 06:59:18.459 | INFO     | __main__:main:1227 - Found 15 repositories
2024-09-13 06:59:18.459 | INFO     | __main__:main:1205 - LeanDojo configured
2024-09-13 06:59:18.459 | INFO     | __main__:main:1227 - Found 15 repositories
2024-09-13 06:59:18.459 | INFO     | __main__:main:1318 - Starting without curriculum learning
2024-09-13 06:59:18.459 | INFO     | __main__:main:1318 - Starting without curriculum learning
2024-09-13 06:59:18.459 | INFO     | __main__:main:1227 - Found 15 repositories
2024-09-13 06:59:18.459 | INFO     | __main__:main:1318 - Starting without curriculum learning
2024-09-13 06:59:18.650 | DEBUG    | lean_dojo.data_extraction.lean:_to_commit_hash:90 - Querying the commit hash for le
an4 v4.7.0
2024-09-13 06:59:18.652 | DEBUG    | lean_dojo.data_extraction.lean:_to_commit_hash:90 - Querying the commit hash for le
an4 v4.7.0
2024-09-13 06:59:18.668 | DEBUG    | lean_dojo.data_extraction.lean:_to_commit_hash:90 - Querying the commit hash for le
an4 v4.7.0
2024-09-13 06:59:34.589 | INFO     | __main__:main:1368 - length of lean_git_repos: 1
2024-09-13 06:59:34.590 | INFO     | __main__:main:1369 - i: 0
2024-09-13 06:59:34.590 | INFO     | __main__:main:1409 - All GPUs
2024-09-13 06:59:34.593 | INFO     | __main__:find_latest_checkpoint:517 - Using the latest checkpoint: /data/yingzi_ma/
lean_project/checkpoints_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0
.1_epoch=1-Recall@10_val=70.07.ckpt
2024-09-13 06:59:34.593 | INFO     | __main__:main:1416 - Found latest checkpoint: /data/yingzi_ma/lean_project/checkpoi
nts_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@10_
val=70.07.ckpt
2024-09-13 06:59:34.593 | INFO     | __main__:main:1422 - Inside train_test_fisher
2024-09-13 06:59:34.593 | INFO     | __main__:main:1423 - Starting training at epoch 1
[rank: 1] Seed set to 3407
2024-09-13 06:59:34.694 | INFO     | __main__:main:1368 - length of lean_git_repos: 1
2024-09-13 06:59:34.695 | INFO     | __main__:main:1369 - i: 0
2024-09-13 06:59:34.695 | INFO     | __main__:main:1409 - All GPUs
2024-09-13 06:59:34.696 | INFO     | __main__:find_latest_checkpoint:517 - Using the latest checkpoint: /data/yingzi_ma/
lean_project/checkpoints_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0
.1_epoch=1-Recall@10_val=70.07.ckpt
2024-09-13 06:59:34.696 | INFO     | __main__:main:1416 - Found latest checkpoint: /data/yingzi_ma/lean_project/checkpoi
nts_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@10_
val=70.07.ckpt
2024-09-13 06:59:34.696 | INFO     | __main__:main:1422 - Inside train_test_fisher
2024-09-13 06:59:34.696 | INFO     | __main__:main:1423 - Starting training at epoch 1
[rank: 3] Seed set to 3407
2024-09-13 06:59:34.730 | INFO     | __main__:main:1368 - length of lean_git_repos: 1
2024-09-13 06:59:34.731 | INFO     | __main__:main:1369 - i: 0
2024-09-13 06:59:34.731 | INFO     | __main__:main:1409 - All GPUs
2024-09-13 06:59:34.731 | INFO     | __main__:find_latest_checkpoint:517 - Using the latest checkpoint: /data/yingzi_ma/
lean_project/checkpoints_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0
.1_epoch=1-Recall@10_val=70.07.ckpt
2024-09-13 06:59:34.732 | INFO     | __main__:main:1416 - Found latest checkpoint: /data/yingzi_ma/lean_project/checkpoi
nts_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@10_
val=70.07.ckpt
2024-09-13 06:59:34.732 | INFO     | __main__:main:1422 - Inside train_test_fisher
2024-09-13 06:59:34.732 | INFO     | __main__:main:1423 - Starting training at epoch 1
[rank: 2] Seed set to 3407
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/lightning_fabric/utilities/cloud_io.py:57: You are
 using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicit
ly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://g
ithub.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default val
ue for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arb
itrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user v
ia `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you
 don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental
feature.
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/lightning_fabric/utilities/cloud_io.py:57: You are
 using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicit
ly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://g
ithub.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default val
ue for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arb
itrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user v
ia `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you
 don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental
feature.
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/lightning_fabric/utilities/cloud_io.py:57: You are
 using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicit
ly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://g
ithub.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default val
ue for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arb
itrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user v
ia `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you
 don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental
feature.
2024-09-13 06:59:37.703 | INFO     | __main__:main:1447 - Loaded premise retriever at /data/yingzi_ma/lean_project/check
points_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@
10_val=70.07.ckpt
2024-09-13 06:59:37.704 | INFO     | __main__:main:1447 - Loaded premise retriever at /data/yingzi_ma/lean_project/check
points_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@
10_val=70.07.ckpt
2024-09-13 06:59:37.704 | INFO     | __main__:main:1447 - Loaded premise retriever at /data/yingzi_ma/lean_project/check
points_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@
10_val=70.07.ckpt
2024-09-13 06:59:37.705 | INFO     | __main__:find_latest_fisher:527 - Using the latest Fisher Information Matrix: /data
/yingzi_ma/lean_project/fisher_PT_single_repo_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_fishe
r_info.pkl
2024-09-13 06:59:37.705 | INFO     | __main__:find_latest_fisher:527 - Using the latest Fisher Information Matrix: /data
/yingzi_ma/lean_project/fisher_PT_single_repo_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_fishe
r_info.pkl
2024-09-13 06:59:37.705 | INFO     | __main__:find_latest_fisher:527 - Using the latest Fisher Information Matrix: /data
/yingzi_ma/lean_project/fisher_PT_single_repo_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_fishe
r_info.pkl
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/storage.py:414: FutureWarning: You are using
 `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It
 is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.
com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for
 `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary
 objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `to
rch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't
 have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental featur
e.
  return torch.load(io.BytesIO(b))
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/storage.py:414: FutureWarning: You are using
 `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It
 is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.
com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for
 `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary
 objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `to
rch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't
 have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental featur
e.
  return torch.load(io.BytesIO(b))
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/storage.py:414: FutureWarning: You are using
 `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It
 is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.
com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for
 `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary
 objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `to
rch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't
 have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental featur
e.
  return torch.load(io.BytesIO(b))
2024-09-13 06:59:38.156 | INFO     | __main__:load_fisher_information:503 - Fisher Information successfully loaded.
2024-09-13 06:59:38.156 | INFO     | retrieval.model:set_fisher_info:60 - Fisher Information has been updated in the mod
el.
2024-09-13 06:59:38.156 | INFO     | __main__:main:1454 - Fisher Information Matrix loaded.
2024-09-13 06:59:38.225 | INFO     | __main__:load_fisher_information:503 - Fisher Information successfully loaded.
2024-09-13 06:59:38.225 | INFO     | retrieval.model:set_fisher_info:60 - Fisher Information has been updated in the mod
el.
2024-09-13 06:59:38.225 | INFO     | __main__:main:1454 - Fisher Information Matrix loaded.
2024-09-13 06:59:38.256 | INFO     | __main__:load_fisher_information:503 - Fisher Information successfully loaded.
2024-09-13 06:59:38.256 | INFO     | retrieval.model:set_fisher_info:60 - Fisher Information has been updated in the mod
el.
2024-09-13 06:59:38.256 | INFO     | __main__:main:1454 - Fisher Information Matrix loaded.
2024-09-13 06:59:38.489 | INFO     | __main__:main:1507 - right before barrier for data module
2024-09-13 06:59:38.489 | INFO     | __main__:main:1507 - right before barrier for data module
2024-09-13 06:59:38.489 | INFO     | __main__:main:1507 - right before barrier for data module
2024-09-13 06:59:38.491 | INFO     | __main__:main:1522 - Data path: /data/yingzi_ma/lean_project/datasets_PT_single_rep
o_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random
2024-09-13 06:59:38.491 | INFO     | __main__:main:1522 - Data path: /data/yingzi_ma/lean_project/datasets_PT_single_rep
o_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random
2024-09-13 06:59:38.491 | INFO     | __main__:main:1522 - Data path: /data/yingzi_ma/lean_project/datasets_PT_single_rep
o_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: Futu
reWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be deprac
ted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github
.com/huggingface/transformers/issues/31884
  warnings.warn(
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: Futu
reWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be deprac
ted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github
.com/huggingface/transformers/issues/31884
  warnings.warn(
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: Futu
reWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be deprac
ted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github
.com/huggingface/transformers/issues/31884
  warnings.warn(
2024-09-13 06:59:38.609 | INFO     | common:__init__:212 - Building the corpus from /data/yingzi_ma/lean_project/dataset
s_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/corpus.jsonl
2024-09-13 06:59:38.609 | INFO     | common:__init__:212 - Building the corpus from /data/yingzi_ma/lean_project/dataset
s_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/corpus.jsonl
2024-09-13 06:59:38.609 | INFO     | common:__init__:212 - Building the corpus from /data/yingzi_ma/lean_project/dataset
s_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/corpus.jsonl
2024-09-13 06:59:49.145 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
Training dataset size: 149228
2024-09-13 06:59:49.163 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_val/cached_data.pkl
Validation dataset size: 2087
2024-09-13 06:59:49.177 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_pred/cached_data.pkl
Testing dataset size: 2298
2024-09-13 06:59:49.177 | INFO     | __main__:main:1536 - Training dataset size after load: 149228
2024-09-13 06:59:49.177 | INFO     | __main__:main:1537 - Validation dataset size after load: 2087
2024-09-13 06:59:49.177 | INFO     | __main__:main:1538 - Testing dataset size after load: 2298
2024-09-13 06:59:49.177 | INFO     | __main__:main:1540 - Starting progressive training from epoch 1 to 2
2024-09-13 06:59:49.177 | INFO     | __main__:main:1543 - hit the barrier before training
2024-09-13 06:59:49.314 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
Training dataset size: 149228
2024-09-13 06:59:49.316 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
Training dataset size: 149228
2024-09-13 06:59:49.326 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_val/cached_data.pkl
Validation dataset size: 2087
2024-09-13 06:59:49.326 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_val/cached_data.pkl
Validation dataset size: 2087
2024-09-13 06:59:49.338 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_pred/cached_data.pkl
Testing dataset size: 2298
2024-09-13 06:59:49.339 | INFO     | __main__:main:1536 - Training dataset size after load: 149228
2024-09-13 06:59:49.339 | INFO     | __main__:main:1537 - Validation dataset size after load: 2087
2024-09-13 06:59:49.339 | INFO     | __main__:main:1538 - Testing dataset size after load: 2298
2024-09-13 06:59:49.339 | INFO     | __main__:main:1540 - Starting progressive training from epoch 1 to 2
2024-09-13 06:59:49.339 | INFO     | __main__:main:1543 - hit the barrier before training
2024-09-13 06:59:49.339 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_pred/cached_data.pkl
Testing dataset size: 2298
2024-09-13 06:59:49.340 | INFO     | __main__:main:1536 - Training dataset size after load: 149228
2024-09-13 06:59:49.340 | INFO     | __main__:main:1537 - Validation dataset size after load: 2087
2024-09-13 06:59:49.340 | INFO     | __main__:main:1538 - Testing dataset size after load: 2298
2024-09-13 06:59:49.340 | INFO     | __main__:main:1540 - Starting progressive training from epoch 1 to 2
2024-09-13 06:59:49.340 | INFO     | __main__:main:1543 - hit the barrier before training
[rank: 3] Seed set to 3407
[rank: 2] Seed set to 3407
[rank: 1] Seed set to 3407
Initializing distributed: GLOBAL_RANK: 2, MEMBER: 3/4
Initializing distributed: GLOBAL_RANK: 3, MEMBER: 4/4
Initializing distributed: GLOBAL_RANK: 1, MEMBER: 2/4
----------------------------------------------------------------------------------------------------
distributed_backend=nccl
All distributed processes registered. Starting with 4 processes
----------------------------------------------------------------------------------------------------

NCCL version 2.20.5+cuda12.4
2024-09-13 06:59:51.787 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
Training dataset size: 149228
2024-09-13 06:59:51.884 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_val/cached_data.pkl
Validation dataset size: 2087
2024-09-13 06:59:51.906 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_pred/cached_data.pkl
Testing dataset size: 2298
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/callbacks/model_checkpoint.py:65
3: Checkpoint directory /data/yingzi_ma/lean_project/checkpoints_PT_single_repo_no_ewc exists and is not empty.
2024-09-13 06:59:53.192 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
2024-09-13 06:59:53.206 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
2024-09-13 06:59:53.267 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
Training dataset size: 149228
Training dataset size: 149228
2024-09-13 06:59:53.294 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_val/cached_data.pkl
2024-09-13 06:59:53.296 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_val/cached_data.pkl
Validation dataset size: 2087
Validation dataset size: 2087
2024-09-13 06:59:53.312 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_pred/cached_data.pkl
2024-09-13 06:59:53.313 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_pred/cached_data.pkl
Testing dataset size: 2298
Testing dataset size: 2298
Training dataset size: 149228
2024-09-13 06:59:53.354 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_val/cached_data.pkl
Validation dataset size: 2087
2024-09-13 06:59:53.369 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_pred/cached_data.pkl
Testing dataset size: 2298
Restoring states from the checkpoint path at /data/yingzi_ma/lean_project/checkpoints_PT_single_repo_no_ewc/merged_with_
new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@10_val=70.07.ckpt
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/lightning_fabric/utilities/cloud_io.py:57: You are
 using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicit
ly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://g
ithub.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default val
ue for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arb
itrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user v
ia `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you
 don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental
feature.
LOCAL_RANK: 3 - CUDA_VISIBLE_DEVICES: [0,1,2,3]
LOCAL_RANK: 1 - CUDA_VISIBLE_DEVICES: [0,1,2,3]
LOCAL_RANK: 2 - CUDA_VISIBLE_DEVICES: [0,1,2,3]
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0,1,2,3]
2024-09-13 06:59:54.433 | INFO     | common:get_optimizers:449 - Optimizing with AdamW
2024-09-13 06:59:54.435 | INFO     | common:get_optimizers:449 - Optimizing with AdamW
2024-09-13 06:59:54.435 | INFO     | common:get_optimizers:449 - Optimizing with AdamW
2024-09-13 06:59:54.436 | INFO     | common:get_optimizers:449 - Optimizing with AdamW

  | Name    | Type           | Params
-------------------------------------------
0 | encoder | T5EncoderModel | 217 M
-------------------------------------------
217 M     Trainable params
0         Non-trainable params
217 M     Total params
870.630   Total estimated model params size (MB)
Restored all states from the checkpoint at /data/yingzi_ma/lean_project/checkpoints_PT_single_repo_no_ewc/merged_with_ne
w_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@10_val=70.07.ckpt
`Trainer.fit` stopped: `max_epochs=2` reached.
2024-09-13 06:59:57.266 | INFO     | __main__:main:1546 - hit the barrier after training
2024-09-13 06:59:57.436 | INFO     | __main__:main:1546 - hit the barrier after training
2024-09-13 06:59:57.523 | INFO     | __main__:main:1546 - hit the barrier after training
2024-09-13 06:59:57.579 | INFO     | __main__:main:1546 - hit the barrier after training
2024-09-13 06:59:57.580 | INFO     | __main__:main:1552 - Finished progressive training at epoch 2
2024-09-13 06:59:57.580 | INFO     | __main__:main:1552 - Finished progressive training at epoch 2
2024-09-13 06:59:57.580 | INFO     | __main__:main:1552 - Finished progressive training at epoch 2
2024-09-13 06:59:57.580 | INFO     | __main__:main:1552 - Finished progressive training at epoch 2
2024-09-13 06:59:57.581 | INFO     | __main__:find_latest_checkpoint:517 - Using the latest checkpoint: /data/yingzi_ma/
lean_project/checkpoints_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0
.1_epoch=1-Recall@10_val=70.07.ckpt
2024-09-13 06:59:57.581 | INFO     | __main__:find_latest_checkpoint:517 - Using the latest checkpoint: /data/yingzi_ma/
lean_project/checkpoints_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0
.1_epoch=1-Recall@10_val=70.07.ckpt
2024-09-13 06:59:57.581 | INFO     | __main__:find_latest_checkpoint:517 - Using the latest checkpoint: /data/yingzi_ma/
lean_project/checkpoints_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0
.1_epoch=1-Recall@10_val=70.07.ckpt
2024-09-13 06:59:57.582 | INFO     | __main__:main:1559 - Found latest checkpoint: /data/yingzi_ma/lean_project/checkpoi
nts_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@10_
val=70.07.ckpt
2024-09-13 06:59:57.581 | INFO     | __main__:find_latest_checkpoint:517 - Using the latest checkpoint: /data/yingzi_ma/
lean_project/checkpoints_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0
.1_epoch=1-Recall@10_val=70.07.ckpt
2024-09-13 06:59:57.582 | INFO     | __main__:main:1559 - Found latest checkpoint: /data/yingzi_ma/lean_project/checkpoi
nts_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@10_
val=70.07.ckpt
2024-09-13 06:59:57.582 | INFO     | __main__:main:1559 - Found latest checkpoint: /data/yingzi_ma/lean_project/checkpoi
nts_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@10_
val=70.07.ckpt
2024-09-13 06:59:57.582 | INFO     | __main__:main:1559 - Found latest checkpoint: /data/yingzi_ma/lean_project/checkpoi
nts_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@10_
val=70.07.ckpt
2024-09-13 06:59:59.677 | INFO     | __main__:main:1625 - Calculating Fisher Information Matrix for EWC
2024-09-13 06:59:59.678 | INFO     | __main__:main:1629 - right before barrier fisher
2024-09-13 06:59:59.761 | INFO     | __main__:main:1625 - Calculating Fisher Information Matrix for EWC
2024-09-13 06:59:59.761 | INFO     | __main__:main:1629 - right before barrier fisher
2024-09-13 06:59:59.782 | INFO     | __main__:main:1625 - Calculating Fisher Information Matrix for EWC
2024-09-13 06:59:59.782 | INFO     | __main__:main:1629 - right before barrier fisher
2024-09-13 06:59:59.796 | INFO     | __main__:main:1625 - Calculating Fisher Information Matrix for EWC
2024-09-13 06:59:59.796 | INFO     | __main__:main:1629 - right before barrier fisher
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/lightning_fabric/plugins/environments/slurm.py:204
: The `srun` command is available on your system but is not used. HINT: If your intention is to run Lightning on SLURM,
prepend your python command with `srun` like so: srun python main.py ...
[rank: 1] Seed set to 3407
[rank: 3] Seed set to 3407
[rank: 0] Seed set to 3407
[rank: 2] Seed set to 3407
2024-09-13 07:00:03.061 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
2024-09-13 07:00:03.078 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
Training dataset size: 149228
2024-09-13 07:00:03.161 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
Training dataset size: 149228
Training dataset size: 149228
2024-09-13 07:00:04.023 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
Training dataset size: 149228
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0,1,2,3]
LOCAL_RANK: 3 - CUDA_VISIBLE_DEVICES: [0,1,2,3]
LOCAL_RANK: 1 - CUDA_VISIBLE_DEVICES: [0,1,2,3]
LOCAL_RANK: 2 - CUDA_VISIBLE_DEVICES: [0,1,2,3]
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/connectors/data_connecto
r.py:232: Using `DistributedSampler` with the dataloaders. During `trainer.test()`, it is recommended to use `Trainer(de
vices=1, num_nodes=1)` to ensure each sample/batch gets evaluated exactly once. Otherwise, multi-device settings use `Di
stributedSampler` that replicates some samples to make sure all devices have same batch size in case of uneven inputs.
Testing DataLoader 0:   0%|                                                                    | 0/9327 [00:00<?, ?it/s]
An error occurred during fisher: element 0 of tensors does not require grad and does not have a grad_fn
Traceback (most recent call last):
  File "/data/yingzi_ma/lean_project/ReProver/main.py", line 1631, in main
    trainer.test(model, datamodule=data_module)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 754, in test
    return call._call_and_handle_interrupt(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/call.py", line 4
3, in _call_and_handle_interrupt
    return trainer.strategy.launcher.launch(trainer_fn, *args, trainer=trainer, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/strategies/launchers/sub
process_script.py", line 105, in launch
    return function(*args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 794, in _test_impl
    results = self._run(model, ckpt_path=ckpt_path)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 987, in _run
    results = self._run_stage()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 1026, in _run_stage
    return self._evaluation_loop.run()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/utilities.py", lin
e 182, in _decorator
    return loop_run(self, *args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/evaluation_loop.py
", line 135, in run
    self._evaluation_step(batch, batch_idx, dataloader_idx, dataloader_iter)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/evaluation_loop.py
", line 396, in _evaluation_step
    output = call._call_strategy_hook(trainer, hook_name, *step_args)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/call.py", line 3
09, in _call_strategy_hook
    output = fn(*args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/strategies/strategy.py",
 line 425, in test_step
    return self.lightning_module.test_step(*args, **kwargs)
  File "/data/yingzi_ma/lean_project/ReProver/retrieval/model.py", line 412, in test_step
    loss.backward()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/_tensor.py", line 521, in backward
    torch.autograd.backward(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/autograd/__init__.py", line 289, in
backward
    _engine_run_backward(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/autograd/graph.py", line 768, in _en
gine_run_backward
    return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn

2024-09-13 07:00:06.278 | INFO     | __main__:main:1692 - Finished processing the repository
2024-09-13 07:00:06.278 | INFO     | __main__:main:1694 - current epoch: 2
2024-09-13 07:00:06.278 | INFO     | __main__:main:1368 - length of lean_git_repos: 1
2024-09-13 07:00:06.278 | INFO     | __main__:main:1369 - i: 1
2024-09-13 07:00:06.279 | INFO     | __main__:main:1696 - An error occurred: list index out of range
Traceback (most recent call last):
  File "/data/yingzi_ma/lean_project/ReProver/main.py", line 1370, in main
    repo = lean_git_repos[i]
IndexError: list index out of range
An error occurred during fisher: element 0 of tensors does not require grad and does not have a grad_fn
Traceback (most recent call last):
  File "/data/yingzi_ma/lean_project/ReProver/main.py", line 1631, in main
    trainer.test(model, datamodule=data_module)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 754, in test
    return call._call_and_handle_interrupt(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/call.py", line 4
3, in _call_and_handle_interrupt
    return trainer.strategy.launcher.launch(trainer_fn, *args, trainer=trainer, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/strategies/launchers/sub
process_script.py", line 105, in launch
    return function(*args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 794, in _test_impl
    results = self._run(model, ckpt_path=ckpt_path)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 987, in _run
    results = self._run_stage()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 1026, in _run_stage
    return self._evaluation_loop.run()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/utilities.py", lin
e 182, in _decorator
    return loop_run(self, *args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/evaluation_loop.py
", line 135, in run
    self._evaluation_step(batch, batch_idx, dataloader_idx, dataloader_iter)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/evaluation_loop.py
", line 396, in _evaluation_step
    output = call._call_strategy_hook(trainer, hook_name, *step_args)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/call.py", line 3
09, in _call_strategy_hook
    output = fn(*args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/strategies/strategy.py",
 line 425, in test_step
    return self.lightning_module.test_step(*args, **kwargs)
  File "/data/yingzi_ma/lean_project/ReProver/retrieval/model.py", line 412, in test_step
    loss.backward()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/_tensor.py", line 521, in backward
    torch.autograd.backward(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/autograd/__init__.py", line 289, in
backward
    _engine_run_backward(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/autograd/graph.py", line 768, in _en
gine_run_backward
    return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn

2024-09-13 07:00:06.479 | INFO     | __main__:main:1692 - Finished processing the repository
2024-09-13 07:00:06.479 | INFO     | __main__:main:1694 - current epoch: 2
2024-09-13 07:00:06.479 | INFO     | __main__:main:1368 - length of lean_git_repos: 1
2024-09-13 07:00:06.479 | INFO     | __main__:main:1369 - i: 1
2024-09-13 07:00:06.479 | INFO     | __main__:main:1696 - An error occurred: list index out of range
Traceback (most recent call last):
  File "/data/yingzi_ma/lean_project/ReProver/main.py", line 1370, in main
    repo = lean_git_repos[i]
IndexError: list index out of range
An error occurred during fisher: element 0 of tensors does not require grad and does not have a grad_fn
Traceback (most recent call last):
  File "/data/yingzi_ma/lean_project/ReProver/main.py", line 1631, in main
    trainer.test(model, datamodule=data_module)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 754, in test
    return call._call_and_handle_interrupt(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/call.py", line 4
3, in _call_and_handle_interrupt
    return trainer.strategy.launcher.launch(trainer_fn, *args, trainer=trainer, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/strategies/launchers/sub
process_script.py", line 105, in launch
    return function(*args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 794, in _test_impl
    results = self._run(model, ckpt_path=ckpt_path)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 987, in _run
    results = self._run_stage()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 1026, in _run_stage
    return self._evaluation_loop.run()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/utilities.py", lin
e 182, in _decorator
    return loop_run(self, *args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/evaluation_loop.py
", line 135, in run
    self._evaluation_step(batch, batch_idx, dataloader_idx, dataloader_iter)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/evaluation_loop.py
", line 396, in _evaluation_step
    output = call._call_strategy_hook(trainer, hook_name, *step_args)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/call.py", line 3
09, in _call_strategy_hook
    output = fn(*args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/strategies/strategy.py",
 line 425, in test_step
    return self.lightning_module.test_step(*args, **kwargs)
  File "/data/yingzi_ma/lean_project/ReProver/retrieval/model.py", line 412, in test_step
    loss.backward()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/_tensor.py", line 521, in backward
    torch.autograd.backward(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/autograd/__init__.py", line 289, in
backward
    _engine_run_backward(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/autograd/graph.py", line 768, in _en
gine_run_backward
    return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn

2024-09-13 07:00:06.566 | INFO     | __main__:main:1692 - Finished processing the repository
2024-09-13 07:00:06.566 | INFO     | __main__:main:1694 - current epoch: 2
2024-09-13 07:00:06.566 | INFO     | __main__:main:1368 - length of lean_git_repos: 1
2024-09-13 07:00:06.567 | INFO     | __main__:main:1369 - i: 1
2024-09-13 07:00:06.567 | INFO     | __main__:main:1696 - An error occurred: list index out of range
Traceback (most recent call last):
  File "/data/yingzi_ma/lean_project/ReProver/main.py", line 1370, in main
    repo = lean_git_repos[i]
IndexError: list index out of range
An error occurred during fisher: element 0 of tensors does not require grad and does not have a grad_fn
Traceback (most recent call last):
  File "/data/yingzi_ma/lean_project/ReProver/main.py", line 1631, in main
    trainer.test(model, datamodule=data_module)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 754, in test
    return call._call_and_handle_interrupt(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/call.py", line 4
3, in _call_and_handle_interrupt
    return trainer.strategy.launcher.launch(trainer_fn, *args, trainer=trainer, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/strategies/launchers/sub
process_script.py", line 105, in launch
    return function(*args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 794, in _test_impl
    results = self._run(model, ckpt_path=ckpt_path)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 987, in _run
    results = self._run_stage()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 1026, in _run_stage
    return self._evaluation_loop.run()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/utilities.py", lin
e 182, in _decorator
    return loop_run(self, *args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/evaluation_loop.py
", line 135, in run
    self._evaluation_step(batch, batch_idx, dataloader_idx, dataloader_iter)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/evaluation_loop.py
", line 396, in _evaluation_step
    output = call._call_strategy_hook(trainer, hook_name, *step_args)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/call.py", line 3
09, in _call_strategy_hook
    output = fn(*args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/strategies/strategy.py",
 line 425, in test_step
    return self.lightning_module.test_step(*args, **kwargs)
  File "/data/yingzi_ma/lean_project/ReProver/retrieval/model.py", line 412, in test_step
    loss.backward()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/_tensor.py", line 521, in backward
    torch.autograd.backward(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/autograd/__init__.py", line 289, in
backward
    _engine_run_backward(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/autograd/graph.py", line 768, in _en
gine_run_backward
    return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn

2024-09-13 07:00:06.582 | INFO     | __main__:main:1692 - Finished processing the repository
2024-09-13 07:00:06.583 | INFO     | __main__:main:1694 - current epoch: 2
2024-09-13 07:00:06.583 | INFO     | __main__:main:1368 - length of lean_git_repos: 1
2024-09-13 07:00:06.583 | INFO     | __main__:main:1369 - i: 1
2024-09-13 07:00:06.583 | INFO     | __main__:main:1696 - An error occurred: list index out of range
Traceback (most recent call last):
  File "/data/yingzi_ma/lean_project/ReProver/main.py", line 1370, in main
    repo = lean_git_repos[i]
IndexError: list index out of range
Testing DataLoader 0:   0%|                                                                    | 0/9327 [00:03<?, ?it/s]
[rank0]:[W913 07:00:09.420119828 ProcessGroupNCCL.cpp:1168] Warning: WARNING: process group has NOT been destroyed befor
e we destruct ProcessGroupNCCL. On normal program exit, the application should call destroy_process_group to ensure that
 any pending NCCL operations have finished in this process. In rare cases this process can exit before this point and bl
ock the progress of another member of the process group. This constraint has always been present,  but this warning has
only been added since PyTorch 2.4 (function operator())
^Z
[8]+  Stopped                 bash run_code.sh
(base) yingzi_ma@compute-permanent-node-352:~/lean_project/ReProver$ bash run_code.sh
Script executed from: /data/yingzi_ma/lean_project/ReProver
Removing old cache files
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834741.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834743.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834746.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834747.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_835354.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_GCS.log': Permission denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834744.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_RAYLET.log': Permission denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834742.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_833907.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_AUTOSCALER.log': Permission deni
ed
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834748.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834745.log': Permiss
ion denied
Stopping ray
Could not terminate `/usr/bin/python3 /usr/local/bin/user_traffic_monitor.py` due to (pid=784996, name='python3')
Did not find any active Ray processes.
Running main.py
[2024-09-13 07:03:09,598] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cuda (auto detect)
2024-09-13 07:03:19.958 | INFO     | __main__:main:1197 - Running progressive training
2024-09-13 07:03:19.958 | INFO     | __main__:main:1203 - Configuring LeanDojo...
2024-09-13 07:03:19.961 | INFO     | generate_benchmark_lean4:configure_leandojo:347 - Current working directory: /data/
yingzi_ma/lean_project/ReProver
2024-09-13 07:03:19.961 | INFO     | __main__:main:1205 - LeanDojo configured
2024-09-13 07:03:19.961 | INFO     | __main__:main:1210 - Starting the main process
2024-09-13 07:03:19.961 | INFO     | __main__:main:1218 - Loading database from /data/yingzi_ma/lean_project/dynamic_dat
abase_PT_single_repo_no_ewc.json
2024-09-13 07:03:40.766 | INFO     | dynamic_database:add_repository:600 - Attempting to add repository: https://github.
com/leanprover-community/mathlib4 (commit: 2b29e73438e240a427bcecc7c0fe19306beb1310)
2024-09-13 07:03:40.766 | INFO     | dynamic_database:add_repository:603 - Added new repository: https://github.com/lean
prover-community/mathlib4 (commit: 2b29e73438e240a427bcecc7c0fe19306beb1310)
2024-09-13 07:03:44.668 | INFO     | dynamic_database:add_repository:600 - Attempting to add repository: https://github.
com/lecopivo/SciLean (commit: 22d53b2f4e3db2a172e71da6eb9c916e62655744)
2024-09-13 07:03:44.668 | INFO     | dynamic_database:add_repository:603 - Added new repository: https://github.com/leco
pivo/SciLean (commit: 22d53b2f4e3db2a172e71da6eb9c916e62655744)
2024-09-13 07:03:44.955 | INFO     | __main__:main:1220 - Loaded database from /data/yingzi_ma/lean_project/dynamic_data
base_PT_single_repo_no_ewc.json
2024-09-13 07:03:44.956 | INFO     | __main__:main:1227 - Found 15 repositories
2024-09-13 07:03:44.956 | INFO     | __main__:main:1318 - Starting without curriculum learning
2024-09-13 07:03:45.339 | DEBUG    | lean_dojo.data_extraction.lean:_to_commit_hash:90 - Querying the commit hash for le
an4 v4.7.0
2024-09-13 07:04:01.588 | INFO     | __main__:main:1347 - Finding compatible repositories...
2024-09-13 07:04:01.600 | INFO     | __main__:main:1350 - Finished finding compatible repositories
2024-09-13 07:04:01.600 | INFO     | __main__:main:1368 - length of lean_git_repos: 1
2024-09-13 07:04:01.600 | INFO     | __main__:main:1369 - i: 0
2024-09-13 07:04:01.601 | INFO     | __main__:main:1375 - Main process
2024-09-13 07:04:01.601 | INFO     | __main__:main:1376 - Using lambda = 0.1
2024-09-13 07:04:01.601 | INFO     | __main__:main:1377 - Processing https://github.com/lecopivo/SciLean
2024-09-13 07:04:01.601 | INFO     | __main__:main:1398 - Adding repo to repos_for_merged_dataset
2024-09-13 07:04:01.601 | INFO     | __main__:main:1409 - All GPUs
2024-09-13 07:04:01.603 | INFO     | __main__:find_latest_checkpoint:517 - Using the latest checkpoint: /data/yingzi_ma/
lean_project/checkpoints_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0
.1_epoch=1-Recall@10_val=70.07.ckpt
2024-09-13 07:04:01.603 | INFO     | __main__:main:1416 - Found latest checkpoint: /data/yingzi_ma/lean_project/checkpoi
nts_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@10_
val=70.07.ckpt
2024-09-13 07:04:01.603 | INFO     | __main__:main:1422 - Inside train_test_fisher
2024-09-13 07:04:01.603 | INFO     | __main__:main:1423 - Starting training at epoch 1
[rank: 0] Seed set to 3407
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/lightning_fabric/utilities/cloud_io.py:57: You are
 using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicit
ly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://g
ithub.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default val
ue for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arb
itrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user v
ia `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you
 don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental
feature.
2024-09-13 07:04:04.477 | INFO     | __main__:main:1447 - Loaded premise retriever at /data/yingzi_ma/lean_project/check
points_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@
10_val=70.07.ckpt
2024-09-13 07:04:04.480 | INFO     | __main__:find_latest_fisher:527 - Using the latest Fisher Information Matrix: /data
/yingzi_ma/lean_project/fisher_PT_single_repo_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_fishe
r_info.pkl
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/storage.py:414: FutureWarning: You are using
 `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It
 is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.
com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for
 `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary
 objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `to
rch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't
 have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental featur
e.
  return torch.load(io.BytesIO(b))
2024-09-13 07:04:04.886 | INFO     | __main__:load_fisher_information:503 - Fisher Information successfully loaded.
2024-09-13 07:04:04.887 | INFO     | retrieval.model:set_fisher_info:60 - Fisher Information has been updated in the mod
el.
2024-09-13 07:04:04.887 | INFO     | __main__:main:1454 - Fisher Information Matrix loaded.
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/lightning_fabric/plugins/environments/slurm.py:204
: The `srun` command is available on your system but is not used. HINT: If your intention is to run Lightning on SLURM,
prepend your python command with `srun` like so: srun python main.py ...
Using bfloat16 Automatic Mixed Precision (AMP)
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
`Trainer(limit_train_batches=1.0)` was configured so 100% of the batches per epoch will be used..
`Trainer(limit_val_batches=1.0)` was configured so 100% of the batches will be used..
2024-09-13 07:04:05.425 | INFO     | __main__:main:1507 - right before barrier for data module
2024-09-13 07:04:05.427 | INFO     | __main__:main:1522 - Data path: /data/yingzi_ma/lean_project/datasets_PT_single_rep
o_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: Futu
reWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be deprac
ted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github
.com/huggingface/transformers/issues/31884
  warnings.warn(
2024-09-13 07:04:05.552 | INFO     | common:__init__:212 - Building the corpus from /data/yingzi_ma/lean_project/dataset
s_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/corpus.jsonl
2024-09-13 07:04:14.632 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
Training dataset size: 149228
2024-09-13 07:04:14.649 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_val/cached_data.pkl
Validation dataset size: 2087
2024-09-13 07:04:14.664 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_pred/cached_data.pkl
Testing dataset size: 2298
2024-09-13 07:04:14.664 | INFO     | __main__:main:1536 - Training dataset size after load: 149228
2024-09-13 07:04:14.664 | INFO     | __main__:main:1537 - Validation dataset size after load: 2087
2024-09-13 07:04:14.664 | INFO     | __main__:main:1538 - Testing dataset size after load: 2298
2024-09-13 07:04:14.664 | INFO     | __main__:main:1540 - Starting progressive training from epoch 1 to 2
2024-09-13 07:04:14.664 | INFO     | __main__:main:1543 - hit the barrier before training
[rank: 0] Seed set to 3407
Initializing distributed: GLOBAL_RANK: 0, MEMBER: 1/4
[2024-09-13 07:04:27,096] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2024-09-13 07:04:27,098] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2024-09-13 07:04:27,098] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cuda (auto detect)
2024-09-13 07:04:36.469 | INFO     | __main__:main:1197 - Running progressive training
2024-09-13 07:04:36.469 | INFO     | __main__:main:1197 - Running progressive training
2024-09-13 07:04:36.469 | INFO     | __main__:main:1197 - Running progressive training
2024-09-13 07:04:36.469 | INFO     | __main__:main:1203 - Configuring LeanDojo...
2024-09-13 07:04:36.469 | INFO     | __main__:main:1203 - Configuring LeanDojo...
2024-09-13 07:04:36.469 | INFO     | __main__:main:1203 - Configuring LeanDojo...
2024-09-13 07:04:36.472 | INFO     | generate_benchmark_lean4:configure_leandojo:347 - Current working directory: /data/
yingzi_ma/lean_project/ReProver
2024-09-13 07:04:36.472 | INFO     | generate_benchmark_lean4:configure_leandojo:347 - Current working directory: /data/
yingzi_ma/lean_project/ReProver
2024-09-13 07:04:36.472 | INFO     | __main__:main:1205 - LeanDojo configured
2024-09-13 07:04:36.472 | INFO     | __main__:main:1205 - LeanDojo configured
2024-09-13 07:04:36.472 | INFO     | generate_benchmark_lean4:configure_leandojo:347 - Current working directory: /data/
yingzi_ma/lean_project/ReProver
2024-09-13 07:04:36.472 | INFO     | __main__:main:1227 - Found 15 repositories
2024-09-13 07:04:36.472 | INFO     | __main__:main:1227 - Found 15 repositories
2024-09-13 07:04:36.472 | INFO     | __main__:main:1318 - Starting without curriculum learning
2024-09-13 07:04:36.472 | INFO     | __main__:main:1205 - LeanDojo configured
2024-09-13 07:04:36.472 | INFO     | __main__:main:1318 - Starting without curriculum learning
2024-09-13 07:04:36.472 | INFO     | __main__:main:1227 - Found 15 repositories
2024-09-13 07:04:36.473 | INFO     | __main__:main:1318 - Starting without curriculum learning
2024-09-13 07:04:36.654 | DEBUG    | lean_dojo.data_extraction.lean:_to_commit_hash:90 - Querying the commit hash for le
an4 v4.7.0
2024-09-13 07:04:36.665 | DEBUG    | lean_dojo.data_extraction.lean:_to_commit_hash:90 - Querying the commit hash for le
an4 v4.7.0
2024-09-13 07:04:36.674 | DEBUG    | lean_dojo.data_extraction.lean:_to_commit_hash:90 - Querying the commit hash for le
an4 v4.7.0
^Z
[9]+  Stopped                 bash run_code.sh
(base) yingzi_ma@compute-permanent-node-352:~/lean_project/ReProver$ bash run_code.sh
Script executed from: /data/yingzi_ma/lean_project/ReProver
Removing old cache files
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834741.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834743.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834746.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834747.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_835354.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_GCS.log': Permission denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834744.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_RAYLET.log': Permission denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834742.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_833907.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_AUTOSCALER.log': Permission deni
ed
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834748.log': Permiss
ion denied
rm: cannot remove '/tmp/ray/session_2024-08-22_03-10-43_407834_833965/logs/events/event_CORE_WORKER_834745.log': Permiss
ion denied
Stopping ray
Could not terminate `/usr/bin/python3 /usr/local/bin/user_traffic_monitor.py` due to (pid=784996, name='python3')
Did not find any active Ray processes.
Running main.py
[2024-09-13 07:05:05,662] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cuda (auto detect)
2024-09-13 07:05:14.965 | INFO     | __main__:main:1197 - Running progressive training
2024-09-13 07:05:14.966 | INFO     | __main__:main:1203 - Configuring LeanDojo...
2024-09-13 07:05:14.968 | INFO     | generate_benchmark_lean4:configure_leandojo:347 - Current working directory: /data/
yingzi_ma/lean_project/ReProver
2024-09-13 07:05:14.968 | INFO     | __main__:main:1205 - LeanDojo configured
2024-09-13 07:05:14.969 | INFO     | __main__:main:1210 - Starting the main process
2024-09-13 07:05:14.969 | INFO     | __main__:main:1218 - Loading database from /data/yingzi_ma/lean_project/dynamic_dat
abase_PT_single_repo_no_ewc.json
2024-09-13 07:05:36.867 | INFO     | dynamic_database:add_repository:600 - Attempting to add repository: https://github.
com/leanprover-community/mathlib4 (commit: 2b29e73438e240a427bcecc7c0fe19306beb1310)
2024-09-13 07:05:36.867 | INFO     | dynamic_database:add_repository:603 - Added new repository: https://github.com/lean
prover-community/mathlib4 (commit: 2b29e73438e240a427bcecc7c0fe19306beb1310)
2024-09-13 07:05:41.125 | INFO     | dynamic_database:add_repository:600 - Attempting to add repository: https://github.
com/lecopivo/SciLean (commit: 22d53b2f4e3db2a172e71da6eb9c916e62655744)
2024-09-13 07:05:41.125 | INFO     | dynamic_database:add_repository:603 - Added new repository: https://github.com/leco
pivo/SciLean (commit: 22d53b2f4e3db2a172e71da6eb9c916e62655744)
2024-09-13 07:05:41.465 | INFO     | __main__:main:1220 - Loaded database from /data/yingzi_ma/lean_project/dynamic_data
base_PT_single_repo_no_ewc.json
2024-09-13 07:05:41.465 | INFO     | __main__:main:1227 - Found 15 repositories
2024-09-13 07:05:41.465 | INFO     | __main__:main:1318 - Starting without curriculum learning
2024-09-13 07:05:41.715 | DEBUG    | lean_dojo.data_extraction.lean:_to_commit_hash:90 - Querying the commit hash for le
an4 v4.7.0
2024-09-13 07:05:57.908 | INFO     | __main__:main:1347 - Finding compatible repositories...
2024-09-13 07:05:57.919 | INFO     | __main__:main:1350 - Finished finding compatible repositories
2024-09-13 07:05:57.920 | INFO     | __main__:main:1368 - length of lean_git_repos: 1
2024-09-13 07:05:57.920 | INFO     | __main__:main:1369 - i: 0
2024-09-13 07:05:57.920 | INFO     | __main__:main:1375 - Main process
2024-09-13 07:05:57.920 | INFO     | __main__:main:1376 - Using lambda = 0.1
2024-09-13 07:05:57.920 | INFO     | __main__:main:1377 - Processing https://github.com/lecopivo/SciLean
2024-09-13 07:05:57.920 | INFO     | __main__:main:1398 - Adding repo to repos_for_merged_dataset
2024-09-13 07:05:57.920 | INFO     | __main__:main:1409 - All GPUs
2024-09-13 07:05:57.922 | INFO     | __main__:find_latest_checkpoint:517 - Using the latest checkpoint: /data/yingzi_ma/
lean_project/checkpoints_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0
.1_epoch=1-Recall@10_val=70.07.ckpt
2024-09-13 07:05:57.922 | INFO     | __main__:main:1416 - Found latest checkpoint: /data/yingzi_ma/lean_project/checkpoi
nts_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@10_
val=70.07.ckpt
2024-09-13 07:05:57.922 | INFO     | __main__:main:1422 - Inside train_test_fisher
2024-09-13 07:05:57.922 | INFO     | __main__:main:1423 - Starting training at epoch 1
[rank: 0] Seed set to 3407
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/lightning_fabric/utilities/cloud_io.py:57: You are
 using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicit
ly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://g
ithub.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default val
ue for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arb
itrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user v
ia `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you
 don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental
feature.
2024-09-13 07:06:00.611 | INFO     | __main__:main:1447 - Loaded premise retriever at /data/yingzi_ma/lean_project/check
points_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@
10_val=70.07.ckpt
2024-09-13 07:06:00.614 | INFO     | __main__:find_latest_fisher:527 - Using the latest Fisher Information Matrix: /data
/yingzi_ma/lean_project/fisher_PT_single_repo_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_fishe
r_info.pkl
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/storage.py:414: FutureWarning: You are using
 `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It
 is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.
com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for
 `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary
 objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `to
rch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't
 have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental featur
e.
  return torch.load(io.BytesIO(b))
2024-09-13 07:06:01.024 | INFO     | __main__:load_fisher_information:503 - Fisher Information successfully loaded.
2024-09-13 07:06:01.024 | INFO     | retrieval.model:set_fisher_info:60 - Fisher Information has been updated in the mod
el.
2024-09-13 07:06:01.025 | INFO     | __main__:main:1454 - Fisher Information Matrix loaded.
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/lightning_fabric/plugins/environments/slurm.py:204
: The `srun` command is available on your system but is not used. HINT: If your intention is to run Lightning on SLURM,
prepend your python command with `srun` like so: srun python main.py ...
Using bfloat16 Automatic Mixed Precision (AMP)
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
`Trainer(limit_val_batches=1)` was configured so 1 batch will be used.
2024-09-13 07:06:01.445 | INFO     | __main__:main:1507 - right before barrier for data module
2024-09-13 07:06:01.445 | INFO     | __main__:main:1522 - Data path: /data/yingzi_ma/lean_project/datasets_PT_single_rep
o_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: Futu
reWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be deprac
ted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github
.com/huggingface/transformers/issues/31884
  warnings.warn(
2024-09-13 07:06:01.589 | INFO     | common:__init__:212 - Building the corpus from /data/yingzi_ma/lean_project/dataset
s_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/corpus.jsonl
2024-09-13 07:06:11.112 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
Training dataset size: 149228
2024-09-13 07:06:11.129 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_val/cached_data.pkl
Validation dataset size: 2087
2024-09-13 07:06:11.144 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_pred/cached_data.pkl
Testing dataset size: 2298
2024-09-13 07:06:11.144 | INFO     | __main__:main:1536 - Training dataset size after load: 149228
2024-09-13 07:06:11.144 | INFO     | __main__:main:1537 - Validation dataset size after load: 2087
2024-09-13 07:06:11.144 | INFO     | __main__:main:1538 - Testing dataset size after load: 2298
2024-09-13 07:06:11.144 | INFO     | __main__:main:1540 - Starting progressive training from epoch 1 to 2
2024-09-13 07:06:11.144 | INFO     | __main__:main:1543 - hit the barrier before training
[rank: 0] Seed set to 3407
Initializing distributed: GLOBAL_RANK: 0, MEMBER: 1/4
[2024-09-13 07:06:23,703] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2024-09-13 07:06:23,703] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2024-09-13 07:06:23,703] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cuda (auto detect)
2024-09-13 07:06:33.876 | INFO     | __main__:main:1197 - Running progressive training
2024-09-13 07:06:33.876 | INFO     | __main__:main:1197 - Running progressive training
2024-09-13 07:06:33.876 | INFO     | __main__:main:1203 - Configuring LeanDojo...
2024-09-13 07:06:33.876 | INFO     | __main__:main:1203 - Configuring LeanDojo...
2024-09-13 07:06:33.876 | INFO     | __main__:main:1197 - Running progressive training
2024-09-13 07:06:33.876 | INFO     | __main__:main:1203 - Configuring LeanDojo...
2024-09-13 07:06:33.880 | INFO     | generate_benchmark_lean4:configure_leandojo:347 - Current working directory: /data/
yingzi_ma/lean_project/ReProver
2024-09-13 07:06:33.880 | INFO     | __main__:main:1205 - LeanDojo configured
2024-09-13 07:06:33.880 | INFO     | __main__:main:1227 - Found 15 repositories
2024-09-13 07:06:33.880 | INFO     | __main__:main:1318 - Starting without curriculum learning
2024-09-13 07:06:33.880 | INFO     | generate_benchmark_lean4:configure_leandojo:347 - Current working directory: /data/
yingzi_ma/lean_project/ReProver
2024-09-13 07:06:33.880 | INFO     | __main__:main:1205 - LeanDojo configured
2024-09-13 07:06:33.880 | INFO     | __main__:main:1227 - Found 15 repositories
2024-09-13 07:06:33.880 | INFO     | __main__:main:1318 - Starting without curriculum learning
2024-09-13 07:06:33.880 | INFO     | generate_benchmark_lean4:configure_leandojo:347 - Current working directory: /data/
yingzi_ma/lean_project/ReProver
2024-09-13 07:06:33.880 | INFO     | __main__:main:1205 - LeanDojo configured
2024-09-13 07:06:33.880 | INFO     | __main__:main:1227 - Found 15 repositories
2024-09-13 07:06:33.880 | INFO     | __main__:main:1318 - Starting without curriculum learning
2024-09-13 07:06:34.067 | DEBUG    | lean_dojo.data_extraction.lean:_to_commit_hash:90 - Querying the commit hash for le
an4 v4.7.0
2024-09-13 07:06:34.100 | DEBUG    | lean_dojo.data_extraction.lean:_to_commit_hash:90 - Querying the commit hash for le
an4 v4.7.0
2024-09-13 07:06:34.107 | DEBUG    | lean_dojo.data_extraction.lean:_to_commit_hash:90 - Querying the commit hash for le
an4 v4.7.0
2024-09-13 07:06:50.166 | INFO     | __main__:main:1368 - length of lean_git_repos: 1
2024-09-13 07:06:50.166 | INFO     | __main__:main:1369 - i: 0
2024-09-13 07:06:50.166 | INFO     | __main__:main:1409 - All GPUs
2024-09-13 07:06:50.170 | INFO     | __main__:find_latest_checkpoint:517 - Using the latest checkpoint: /data/yingzi_ma/
lean_project/checkpoints_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0
.1_epoch=1-Recall@10_val=70.07.ckpt
2024-09-13 07:06:50.170 | INFO     | __main__:main:1416 - Found latest checkpoint: /data/yingzi_ma/lean_project/checkpoi
nts_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@10_
val=70.07.ckpt
2024-09-13 07:06:50.170 | INFO     | __main__:main:1422 - Inside train_test_fisher
2024-09-13 07:06:50.170 | INFO     | __main__:main:1423 - Starting training at epoch 1
[rank: 3] Seed set to 3407
2024-09-13 07:06:50.294 | INFO     | __main__:main:1368 - length of lean_git_repos: 1
2024-09-13 07:06:50.294 | INFO     | __main__:main:1369 - i: 0
2024-09-13 07:06:50.294 | INFO     | __main__:main:1409 - All GPUs
2024-09-13 07:06:50.295 | INFO     | __main__:find_latest_checkpoint:517 - Using the latest checkpoint: /data/yingzi_ma/
lean_project/checkpoints_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0
.1_epoch=1-Recall@10_val=70.07.ckpt
2024-09-13 07:06:50.295 | INFO     | __main__:main:1416 - Found latest checkpoint: /data/yingzi_ma/lean_project/checkpoi
nts_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@10_
val=70.07.ckpt
2024-09-13 07:06:50.295 | INFO     | __main__:main:1422 - Inside train_test_fisher
2024-09-13 07:06:50.295 | INFO     | __main__:main:1423 - Starting training at epoch 1
[rank: 2] Seed set to 3407
2024-09-13 07:06:50.352 | INFO     | __main__:main:1368 - length of lean_git_repos: 1
2024-09-13 07:06:50.352 | INFO     | __main__:main:1369 - i: 0
2024-09-13 07:06:50.353 | INFO     | __main__:main:1409 - All GPUs
2024-09-13 07:06:50.354 | INFO     | __main__:find_latest_checkpoint:517 - Using the latest checkpoint: /data/yingzi_ma/
lean_project/checkpoints_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0
.1_epoch=1-Recall@10_val=70.07.ckpt
2024-09-13 07:06:50.354 | INFO     | __main__:main:1416 - Found latest checkpoint: /data/yingzi_ma/lean_project/checkpoi
nts_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@10_
val=70.07.ckpt
2024-09-13 07:06:50.354 | INFO     | __main__:main:1422 - Inside train_test_fisher
2024-09-13 07:06:50.354 | INFO     | __main__:main:1423 - Starting training at epoch 1
[rank: 1] Seed set to 3407
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/lightning_fabric/utilities/cloud_io.py:57: You are
 using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicit
ly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://g
ithub.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default val
ue for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arb
itrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user v
ia `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you
 don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental
feature.
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/lightning_fabric/utilities/cloud_io.py:57: You are
 using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicit
ly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://g
ithub.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default val
ue for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arb
itrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user v
ia `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you
 don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental
feature.
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/lightning_fabric/utilities/cloud_io.py:57: You are
 using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicit
ly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://g
ithub.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default val
ue for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arb
itrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user v
ia `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you
 don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental
feature.
2024-09-13 07:06:53.644 | INFO     | __main__:main:1447 - Loaded premise retriever at /data/yingzi_ma/lean_project/check
points_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@
10_val=70.07.ckpt
2024-09-13 07:06:53.644 | INFO     | __main__:main:1447 - Loaded premise retriever at /data/yingzi_ma/lean_project/check
points_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@
10_val=70.07.ckpt
2024-09-13 07:06:53.648 | INFO     | __main__:find_latest_fisher:527 - Using the latest Fisher Information Matrix: /data
/yingzi_ma/lean_project/fisher_PT_single_repo_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_fishe
r_info.pkl
2024-09-13 07:06:53.648 | INFO     | __main__:find_latest_fisher:527 - Using the latest Fisher Information Matrix: /data
/yingzi_ma/lean_project/fisher_PT_single_repo_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_fishe
r_info.pkl
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/storage.py:414: FutureWarning: You are using
 `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It
 is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.
com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for
 `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary
 objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `to
rch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't
 have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental featur
e.
  return torch.load(io.BytesIO(b))
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/storage.py:414: FutureWarning: You are using
 `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It
 is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.
com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for
 `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary
 objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `to
rch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't
 have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental featur
e.
  return torch.load(io.BytesIO(b))
2024-09-13 07:06:53.659 | INFO     | __main__:main:1447 - Loaded premise retriever at /data/yingzi_ma/lean_project/check
points_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@
10_val=70.07.ckpt
2024-09-13 07:06:53.661 | INFO     | __main__:find_latest_fisher:527 - Using the latest Fisher Information Matrix: /data
/yingzi_ma/lean_project/fisher_PT_single_repo_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_fishe
r_info.pkl
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/storage.py:414: FutureWarning: You are using
 `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It
 is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.
com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for
 `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary
 objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `to
rch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't
 have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental featur
e.
  return torch.load(io.BytesIO(b))
2024-09-13 07:06:54.240 | INFO     | __main__:load_fisher_information:503 - Fisher Information successfully loaded.
2024-09-13 07:06:54.240 | INFO     | retrieval.model:set_fisher_info:60 - Fisher Information has been updated in the mod
el.
2024-09-13 07:06:54.240 | INFO     | __main__:main:1454 - Fisher Information Matrix loaded.
2024-09-13 07:06:54.246 | INFO     | __main__:load_fisher_information:503 - Fisher Information successfully loaded.
2024-09-13 07:06:54.246 | INFO     | retrieval.model:set_fisher_info:60 - Fisher Information has been updated in the mod
el.
2024-09-13 07:06:54.246 | INFO     | __main__:main:1454 - Fisher Information Matrix loaded.
2024-09-13 07:06:54.261 | INFO     | __main__:load_fisher_information:503 - Fisher Information successfully loaded.
2024-09-13 07:06:54.261 | INFO     | retrieval.model:set_fisher_info:60 - Fisher Information has been updated in the mod
el.
2024-09-13 07:06:54.261 | INFO     | __main__:main:1454 - Fisher Information Matrix loaded.
2024-09-13 07:06:54.673 | INFO     | __main__:main:1507 - right before barrier for data module
2024-09-13 07:06:54.673 | INFO     | __main__:main:1507 - right before barrier for data module
2024-09-13 07:06:54.674 | INFO     | __main__:main:1507 - right before barrier for data module
2024-09-13 07:06:54.674 | INFO     | __main__:main:1522 - Data path: /data/yingzi_ma/lean_project/datasets_PT_single_rep
o_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random
2024-09-13 07:06:54.674 | INFO     | __main__:main:1522 - Data path: /data/yingzi_ma/lean_project/datasets_PT_single_rep
o_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random
2024-09-13 07:06:54.674 | INFO     | __main__:main:1522 - Data path: /data/yingzi_ma/lean_project/datasets_PT_single_rep
o_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: Futu
reWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be deprac
ted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github
.com/huggingface/transformers/issues/31884
  warnings.warn(
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: Futu
reWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be deprac
ted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github
.com/huggingface/transformers/issues/31884
  warnings.warn(
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: Futu
reWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be deprac
ted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github
.com/huggingface/transformers/issues/31884
  warnings.warn(
2024-09-13 07:06:54.805 | INFO     | common:__init__:212 - Building the corpus from /data/yingzi_ma/lean_project/dataset
s_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/corpus.jsonl
2024-09-13 07:06:54.805 | INFO     | common:__init__:212 - Building the corpus from /data/yingzi_ma/lean_project/dataset
s_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/corpus.jsonl
2024-09-13 07:06:54.805 | INFO     | common:__init__:212 - Building the corpus from /data/yingzi_ma/lean_project/dataset
s_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/corpus.jsonl
2024-09-13 07:07:05.662 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
Training dataset size: 149228
2024-09-13 07:07:05.681 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_val/cached_data.pkl
Validation dataset size: 2087
2024-09-13 07:07:05.696 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_pred/cached_data.pkl
Testing dataset size: 2298
2024-09-13 07:07:05.696 | INFO     | __main__:main:1536 - Training dataset size after load: 149228
2024-09-13 07:07:05.696 | INFO     | __main__:main:1537 - Validation dataset size after load: 2087
2024-09-13 07:07:05.696 | INFO     | __main__:main:1538 - Testing dataset size after load: 2298
2024-09-13 07:07:05.696 | INFO     | __main__:main:1540 - Starting progressive training from epoch 1 to 2
2024-09-13 07:07:05.696 | INFO     | __main__:main:1543 - hit the barrier before training
2024-09-13 07:07:05.806 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
Training dataset size: 149228
2024-09-13 07:07:05.819 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_val/cached_data.pkl
Validation dataset size: 2087
2024-09-13 07:07:05.831 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_pred/cached_data.pkl
Testing dataset size: 2298
2024-09-13 07:07:05.831 | INFO     | __main__:main:1536 - Training dataset size after load: 149228
2024-09-13 07:07:05.831 | INFO     | __main__:main:1537 - Validation dataset size after load: 2087
2024-09-13 07:07:05.831 | INFO     | __main__:main:1538 - Testing dataset size after load: 2298
2024-09-13 07:07:05.831 | INFO     | __main__:main:1540 - Starting progressive training from epoch 1 to 2
2024-09-13 07:07:05.831 | INFO     | __main__:main:1543 - hit the barrier before training
[rank: 1] Seed set to 3407
[rank: 3] Seed set to 3407
Initializing distributed: GLOBAL_RANK: 3, MEMBER: 4/4
Initializing distributed: GLOBAL_RANK: 1, MEMBER: 2/4
2024-09-13 07:07:06.185 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
Training dataset size: 149228
2024-09-13 07:07:06.197 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_val/cached_data.pkl
Validation dataset size: 2087
2024-09-13 07:07:06.209 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_pred/cached_data.pkl
Testing dataset size: 2298
2024-09-13 07:07:06.209 | INFO     | __main__:main:1536 - Training dataset size after load: 149228
2024-09-13 07:07:06.209 | INFO     | __main__:main:1537 - Validation dataset size after load: 2087
2024-09-13 07:07:06.209 | INFO     | __main__:main:1538 - Testing dataset size after load: 2298
2024-09-13 07:07:06.209 | INFO     | __main__:main:1540 - Starting progressive training from epoch 1 to 2
2024-09-13 07:07:06.209 | INFO     | __main__:main:1543 - hit the barrier before training
[rank: 2] Seed set to 3407
Initializing distributed: GLOBAL_RANK: 2, MEMBER: 3/4
----------------------------------------------------------------------------------------------------
distributed_backend=nccl
All distributed processes registered. Starting with 4 processes
----------------------------------------------------------------------------------------------------

NCCL version 2.20.5+cuda12.4
2024-09-13 07:07:08.598 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
Training dataset size: 149228
2024-09-13 07:07:08.711 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_val/cached_data.pkl
Validation dataset size: 2087
2024-09-13 07:07:08.731 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_pred/cached_data.pkl
Testing dataset size: 2298
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/callbacks/model_checkpoint.py:65
3: Checkpoint directory /data/yingzi_ma/lean_project/checkpoints_PT_single_repo_no_ewc exists and is not empty.
2024-09-13 07:07:10.055 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
2024-09-13 07:07:10.060 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
2024-09-13 07:07:10.111 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
Training dataset size: 149228
Training dataset size: 149228
2024-09-13 07:07:10.149 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_val/cached_data.pkl
Validation dataset size: 2087
2024-09-13 07:07:10.153 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_val/cached_data.pkl
Validation dataset size: 2087
2024-09-13 07:07:10.165 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_pred/cached_data.pkl
Testing dataset size: 2298
2024-09-13 07:07:10.174 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_pred/cached_data.pkl
Testing dataset size: 2298
Training dataset size: 149228
2024-09-13 07:07:10.195 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_val/cached_data.pkl
Validation dataset size: 2087
2024-09-13 07:07:10.210 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_pred/cached_data.pkl
Testing dataset size: 2298
Restoring states from the checkpoint path at /data/yingzi_ma/lean_project/checkpoints_PT_single_repo_no_ewc/merged_with_
new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@10_val=70.07.ckpt
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/lightning_fabric/utilities/cloud_io.py:57: You are
 using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicit
ly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://g
ithub.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default val
ue for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arb
itrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user v
ia `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you
 don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental
feature.
LOCAL_RANK: 2 - CUDA_VISIBLE_DEVICES: [0,1,2,3]
LOCAL_RANK: 1 - CUDA_VISIBLE_DEVICES: [0,1,2,3]
LOCAL_RANK: 3 - CUDA_VISIBLE_DEVICES: [0,1,2,3]
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0,1,2,3]
2024-09-13 07:07:11.405 | INFO     | common:get_optimizers:449 - Optimizing with AdamW
2024-09-13 07:07:11.407 | INFO     | common:get_optimizers:449 - Optimizing with AdamW
2024-09-13 07:07:11.408 | INFO     | common:get_optimizers:449 - Optimizing with AdamW
2024-09-13 07:07:11.410 | INFO     | common:get_optimizers:449 - Optimizing with AdamW

  | Name    | Type           | Params
-------------------------------------------
0 | encoder | T5EncoderModel | 217 M
-------------------------------------------
217 M     Trainable params
0         Non-trainable params
217 M     Total params
870.630   Total estimated model params size (MB)
Restored all states from the checkpoint at /data/yingzi_ma/lean_project/checkpoints_PT_single_repo_no_ewc/merged_with_ne
w_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@10_val=70.07.ckpt
`Trainer.fit` stopped: `max_epochs=2` reached.
2024-09-13 07:07:14.994 | INFO     | __main__:main:1546 - hit the barrier after training
2024-09-13 07:07:15.242 | INFO     | __main__:main:1546 - hit the barrier after training
2024-09-13 07:07:15.254 | INFO     | __main__:main:1546 - hit the barrier after training
2024-09-13 07:07:15.257 | INFO     | __main__:main:1546 - hit the barrier after training
2024-09-13 07:07:15.257 | INFO     | __main__:main:1552 - Finished progressive training at epoch 2
2024-09-13 07:07:15.257 | INFO     | __main__:main:1552 - Finished progressive training at epoch 2
2024-09-13 07:07:15.257 | INFO     | __main__:main:1552 - Finished progressive training at epoch 2
2024-09-13 07:07:15.257 | INFO     | __main__:main:1552 - Finished progressive training at epoch 2
2024-09-13 07:07:15.260 | INFO     | __main__:find_latest_checkpoint:517 - Using the latest checkpoint: /data/yingzi_ma/
lean_project/checkpoints_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0
.1_epoch=1-Recall@10_val=70.07.ckpt
2024-09-13 07:07:15.260 | INFO     | __main__:find_latest_checkpoint:517 - Using the latest checkpoint: /data/yingzi_ma/
lean_project/checkpoints_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0
.1_epoch=1-Recall@10_val=70.07.ckpt
2024-09-13 07:07:15.260 | INFO     | __main__:main:1559 - Found latest checkpoint: /data/yingzi_ma/lean_project/checkpoi
nts_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@10_
val=70.07.ckpt
2024-09-13 07:07:15.260 | INFO     | __main__:main:1559 - Found latest checkpoint: /data/yingzi_ma/lean_project/checkpoi
nts_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@10_
val=70.07.ckpt
2024-09-13 07:07:15.260 | INFO     | __main__:find_latest_checkpoint:517 - Using the latest checkpoint: /data/yingzi_ma/
lean_project/checkpoints_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0
.1_epoch=1-Recall@10_val=70.07.ckpt
2024-09-13 07:07:15.260 | INFO     | __main__:find_latest_checkpoint:517 - Using the latest checkpoint: /data/yingzi_ma/
lean_project/checkpoints_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0
.1_epoch=1-Recall@10_val=70.07.ckpt
2024-09-13 07:07:15.260 | INFO     | __main__:main:1559 - Found latest checkpoint: /data/yingzi_ma/lean_project/checkpoi
nts_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@10_
val=70.07.ckpt
2024-09-13 07:07:15.260 | INFO     | __main__:main:1559 - Found latest checkpoint: /data/yingzi_ma/lean_project/checkpoi
nts_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744_lambda_0.1_epoch=1-Recall@10_
val=70.07.ckpt
2024-09-13 07:07:17.669 | INFO     | __main__:main:1625 - Calculating Fisher Information Matrix for EWC
2024-09-13 07:07:17.669 | INFO     | __main__:main:1629 - right before barrier fisher
2024-09-13 07:07:17.697 | INFO     | __main__:main:1625 - Calculating Fisher Information Matrix for EWC
2024-09-13 07:07:17.697 | INFO     | __main__:main:1629 - right before barrier fisher
2024-09-13 07:07:18.106 | INFO     | __main__:main:1625 - Calculating Fisher Information Matrix for EWC
2024-09-13 07:07:18.106 | INFO     | __main__:main:1629 - right before barrier fisher
2024-09-13 07:07:18.130 | INFO     | __main__:main:1625 - Calculating Fisher Information Matrix for EWC
2024-09-13 07:07:18.130 | INFO     | __main__:main:1629 - right before barrier fisher
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/lightning_fabric/plugins/environments/slurm.py:204
: The `srun` command is available on your system but is not used. HINT: If your intention is to run Lightning on SLURM,
prepend your python command with `srun` like so: srun python main.py ...
[rank: 1] Seed set to 3407
[rank: 0] Seed set to 3407
[rank: 2] Seed set to 3407
[rank: 3] Seed set to 3407
2024-09-13 07:07:21.470 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
Training dataset size: 149228
2024-09-13 07:07:23.004 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
Training dataset size: 149228
2024-09-13 07:07:23.297 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
2024-09-13 07:07:23.324 | INFO     | retrieval.datamodule:load_or_cache_data:52 - Loaded data from cache /data/yingzi_ma
/lean_project/datasets_PT_single_repo_no_ewc/merged_with_new_SciLean_22d53b2f4e3db2a172e71da6eb9c916e62655744/random/cac
he_train/cached_data.pkl
Training dataset size: 149228
Training dataset size: 149228
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0,1,2,3]
LOCAL_RANK: 1 - CUDA_VISIBLE_DEVICES: [0,1,2,3]
LOCAL_RANK: 3 - CUDA_VISIBLE_DEVICES: [0,1,2,3]
LOCAL_RANK: 2 - CUDA_VISIBLE_DEVICES: [0,1,2,3]
/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/connectors/data_connecto
r.py:232: Using `DistributedSampler` with the dataloaders. During `trainer.test()`, it is recommended to use `Trainer(de
vices=1, num_nodes=1)` to ensure each sample/batch gets evaluated exactly once. Otherwise, multi-device settings use `Di
stributedSampler` that replicates some samples to make sure all devices have same batch size in case of uneven inputs.
2024-09-13 07:07:24.164 | INFO     | retrieval.model:on_test_start:398 - Fisher info cleared
2024-09-13 07:07:24.193 | INFO     | retrieval.model:on_test_start:400 - Previous params saved
2024-09-13 07:07:24.242 | INFO     | retrieval.model:on_test_start:398 - Fisher info cleared
2024-09-13 07:07:24.265 | INFO     | retrieval.model:test_step:406 - Inside test_step
2024-09-13 07:07:24.284 | INFO     | retrieval.model:on_test_start:400 - Previous params saved
2024-09-13 07:07:24.336 | INFO     | retrieval.model:on_test_start:398 - Fisher info cleared
Testing: |                                                                                        | 0/? [00:00<?, ?it/s]
2024-09-13 07:07:24.361 | INFO     | retrieval.model:on_test_start:398 - Fisher info cleared
2024-09-13 07:07:24.392 | INFO     | retrieval.model:on_test_start:400 - Previous params saved
2024-09-13 07:07:24.394 | INFO     | retrieval.model:on_test_start:400 - Previous params saved
Testing DataLoader 0:   0%|                                                                    | 0/9327 [00:00<?, ?it/s]
2024-09-13 07:07:24.429 | INFO     | retrieval.model:test_step:406 - Inside test_step
2024-09-13 07:07:24.451 | INFO     | retrieval.model:test_step:406 - Inside test_step
2024-09-13 07:07:24.470 | INFO     | retrieval.model:test_step:406 - Inside test_step
2024-09-13 07:07:24.833 | INFO     | retrieval.model:test_step:416 - Test loss before EWC: 0.0302
2024-09-13 07:07:24.972 | INFO     | retrieval.model:test_step:416 - Test loss before EWC: 0.0136
2024-09-13 07:07:25.037 | INFO     | retrieval.model:test_step:416 - Test loss before EWC: 0.0247
2024-09-13 07:07:25.048 | INFO     | retrieval.model:test_step:416 - Test loss before EWC: 0.0354
An error occurred during fisher: element 0 of tensors does not require grad and does not have a grad_fn
Traceback (most recent call last):
  File "/data/yingzi_ma/lean_project/ReProver/main.py", line 1631, in main
    trainer.test(model, datamodule=data_module)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 754, in test
    return call._call_and_handle_interrupt(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/call.py", line 4
3, in _call_and_handle_interrupt
    return trainer.strategy.launcher.launch(trainer_fn, *args, trainer=trainer, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/strategies/launchers/sub
process_script.py", line 105, in launch
    return function(*args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 794, in _test_impl
    results = self._run(model, ckpt_path=ckpt_path)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 987, in _run
    results = self._run_stage()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 1026, in _run_stage
    return self._evaluation_loop.run()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/utilities.py", lin
e 182, in _decorator
    return loop_run(self, *args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/evaluation_loop.py
", line 135, in run
    self._evaluation_step(batch, batch_idx, dataloader_idx, dataloader_iter)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/evaluation_loop.py
", line 396, in _evaluation_step
    output = call._call_strategy_hook(trainer, hook_name, *step_args)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/call.py", line 3
09, in _call_strategy_hook
    output = fn(*args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/strategies/strategy.py",
 line 425, in test_step
    return self.lightning_module.test_step(*args, **kwargs)
  File "/data/yingzi_ma/lean_project/ReProver/retrieval/model.py", line 417, in test_step
    loss.backward()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/_tensor.py", line 521, in backward
    torch.autograd.backward(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/autograd/__init__.py", line 289, in
backward
    _engine_run_backward(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/autograd/graph.py", line 768, in _en
gine_run_backward
    return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn

2024-09-13 07:07:25.709 | INFO     | __main__:main:1692 - Finished processing the repository
2024-09-13 07:07:25.709 | INFO     | __main__:main:1694 - current epoch: 2
2024-09-13 07:07:25.709 | INFO     | __main__:main:1368 - length of lean_git_repos: 1
2024-09-13 07:07:25.709 | INFO     | __main__:main:1369 - i: 1
2024-09-13 07:07:25.709 | INFO     | __main__:main:1696 - An error occurred: list index out of range
Traceback (most recent call last):
  File "/data/yingzi_ma/lean_project/ReProver/main.py", line 1370, in main
    repo = lean_git_repos[i]
IndexError: list index out of range
An error occurred during fisher: element 0 of tensors does not require grad and does not have a grad_fn
Traceback (most recent call last):
  File "/data/yingzi_ma/lean_project/ReProver/main.py", line 1631, in main
    trainer.test(model, datamodule=data_module)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 754, in test
    return call._call_and_handle_interrupt(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/call.py", line 4
3, in _call_and_handle_interrupt
    return trainer.strategy.launcher.launch(trainer_fn, *args, trainer=trainer, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/strategies/launchers/sub
process_script.py", line 105, in launch
    return function(*args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 794, in _test_impl
    results = self._run(model, ckpt_path=ckpt_path)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 987, in _run
    results = self._run_stage()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 1026, in _run_stage
    return self._evaluation_loop.run()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/utilities.py", lin
e 182, in _decorator
    return loop_run(self, *args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/evaluation_loop.py
", line 135, in run
    self._evaluation_step(batch, batch_idx, dataloader_idx, dataloader_iter)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/evaluation_loop.py
", line 396, in _evaluation_step
    output = call._call_strategy_hook(trainer, hook_name, *step_args)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/call.py", line 3
09, in _call_strategy_hook
    output = fn(*args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/strategies/strategy.py",
 line 425, in test_step
    return self.lightning_module.test_step(*args, **kwargs)
  File "/data/yingzi_ma/lean_project/ReProver/retrieval/model.py", line 417, in test_step
    loss.backward()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/_tensor.py", line 521, in backward
    torch.autograd.backward(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/autograd/__init__.py", line 289, in
backward
    _engine_run_backward(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/autograd/graph.py", line 768, in _en
gine_run_backward
    return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn

2024-09-13 07:07:25.901 | INFO     | __main__:main:1692 - Finished processing the repository
2024-09-13 07:07:25.901 | INFO     | __main__:main:1694 - current epoch: 2
2024-09-13 07:07:25.902 | INFO     | __main__:main:1368 - length of lean_git_repos: 1
2024-09-13 07:07:25.902 | INFO     | __main__:main:1369 - i: 1
2024-09-13 07:07:25.902 | INFO     | __main__:main:1696 - An error occurred: list index out of range
Traceback (most recent call last):
  File "/data/yingzi_ma/lean_project/ReProver/main.py", line 1370, in main
    repo = lean_git_repos[i]
IndexError: list index out of range
An error occurred during fisher: element 0 of tensors does not require grad and does not have a grad_fn
Traceback (most recent call last):
  File "/data/yingzi_ma/lean_project/ReProver/main.py", line 1631, in main
    trainer.test(model, datamodule=data_module)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 754, in test
    return call._call_and_handle_interrupt(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/call.py", line 4
3, in _call_and_handle_interrupt
    return trainer.strategy.launcher.launch(trainer_fn, *args, trainer=trainer, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/strategies/launchers/sub
process_script.py", line 105, in launch
    return function(*args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 794, in _test_impl
    results = self._run(model, ckpt_path=ckpt_path)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 987, in _run
    results = self._run_stage()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 1026, in _run_stage
    return self._evaluation_loop.run()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/utilities.py", lin
e 182, in _decorator
    return loop_run(self, *args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/evaluation_loop.py
", line 135, in run
    self._evaluation_step(batch, batch_idx, dataloader_idx, dataloader_iter)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/evaluation_loop.py
", line 396, in _evaluation_step
    output = call._call_strategy_hook(trainer, hook_name, *step_args)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/call.py", line 3
09, in _call_strategy_hook
    output = fn(*args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/strategies/strategy.py",
 line 425, in test_step
    return self.lightning_module.test_step(*args, **kwargs)
  File "/data/yingzi_ma/lean_project/ReProver/retrieval/model.py", line 417, in test_step
    loss.backward()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/_tensor.py", line 521, in backward
    torch.autograd.backward(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/autograd/__init__.py", line 289, in
backward
    _engine_run_backward(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/autograd/graph.py", line 768, in _en
gine_run_backward
    return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn

2024-09-13 07:07:25.935 | INFO     | __main__:main:1692 - Finished processing the repository
2024-09-13 07:07:25.935 | INFO     | __main__:main:1694 - current epoch: 2
2024-09-13 07:07:25.935 | INFO     | __main__:main:1368 - length of lean_git_repos: 1
2024-09-13 07:07:25.935 | INFO     | __main__:main:1369 - i: 1
2024-09-13 07:07:25.935 | INFO     | __main__:main:1696 - An error occurred: list index out of range
Traceback (most recent call last):
  File "/data/yingzi_ma/lean_project/ReProver/main.py", line 1370, in main
    repo = lean_git_repos[i]
IndexError: list index out of range
An error occurred during fisher: element 0 of tensors does not require grad and does not have a grad_fn
Traceback (most recent call last):
  File "/data/yingzi_ma/lean_project/ReProver/main.py", line 1631, in main
    trainer.test(model, datamodule=data_module)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 754, in test
    return call._call_and_handle_interrupt(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/call.py", line 4
3, in _call_and_handle_interrupt
    return trainer.strategy.launcher.launch(trainer_fn, *args, trainer=trainer, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/strategies/launchers/sub
process_script.py", line 105, in launch
    return function(*args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 794, in _test_impl
    results = self._run(model, ckpt_path=ckpt_path)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 987, in _run
    results = self._run_stage()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", lin
e 1026, in _run_stage
    return self._evaluation_loop.run()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/utilities.py", lin
e 182, in _decorator
    return loop_run(self, *args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/evaluation_loop.py
", line 135, in run
    self._evaluation_step(batch, batch_idx, dataloader_idx, dataloader_iter)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/loops/evaluation_loop.py
", line 396, in _evaluation_step
    output = call._call_strategy_hook(trainer, hook_name, *step_args)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/trainer/call.py", line 3
09, in _call_strategy_hook
    output = fn(*args, **kwargs)
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/pytorch_lightning/strategies/strategy.py",
 line 425, in test_step
    return self.lightning_module.test_step(*args, **kwargs)
  File "/data/yingzi_ma/lean_project/ReProver/retrieval/model.py", line 417, in test_step
    loss.backward()
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/_tensor.py", line 521, in backward
    torch.autograd.backward(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/autograd/__init__.py", line 289, in
backward
    _engine_run_backward(
  File "/data/yingzi_ma/miniconda3/envs/ReProver/lib/python3.10/site-packages/torch/autograd/graph.py", line 768, in _en
gine_run_backward
    return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn

2024-09-13 07:07:26.047 | INFO     | __main__:main:1692 - Finished processing the repository
2024-09-13 07:07:26.047 | INFO     | __main__:main:1694 - current epoch: 2
2024-09-13 07:07:26.048 | INFO     | __main__:main:1368 - length of lean_git_repos: 1
2024-09-13 07:07:26.048 | INFO     | __main__:main:1369 - i: 1
2024-09-13 07:07:26.048 | INFO     | __main__:main:1696 - An error occurred: list index out of range
Traceback (most recent call last):
  File "/data/yingzi_ma/lean_project/ReProver/main.py", line 1370, in main
    repo = lean_git_repos[i]
IndexError: list index out of range
Testing DataLoader 0:   0%|                                                                    | 0/9327 [00:03<?, ?it/s]
[rank0]:[W913 07:07:29.362491251 ProcessGroupNCCL.cpp:1168] Warning: WARNING: process group has NOT been destroyed befor
e we destruct ProcessGroupNCCL. On normal program exit, the application should call destroy_process_group to ensure that
 any pending NCCL operations have finished in this process. In rare cases this process can exit before this point and bl
ock the progress of another member of the process group. This constraint has always been present,  but this warning has
only been added since PyTorch 2.4 (function operator())
(base) yingzi_ma@compute-permanent-node-352:~/lean_project/ReProver$
